nabu-log-exporter/src/main/java/com/github/millerm/NabuLogExporter.java

```diff
@@ -0,0 +1,261 @@
+                if (Files.isRegularFile(entry)) {
+                    if (Files.size(entry) < MAX_LOG_FILE_SIZE) {
+                        executorService.submit(new LogTask(entry, stateFilePath));
+                    }
+
+                }
+            }
+        } catch (IOException e) {
+            LOG.severe("Error reading existing files: " + e.getMessage());
+        }
+
+        try (WatchService watchService = FileSystems.getDefault().newWatchService()) {
+            LOG_PATH_DIR.register(watchService, StandardWatchEventKinds.ENTRY_CREATE);
+
+            LOG.info("Watching for new log files at: " + LOG_PATH_DIR);
+
+            while (true) {
+                WatchKey key;
+                try {
+                    key = watchService.take();
+                } catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
+                    LOG.severe("Directory watching interrupted");
+                    return;
+                }
+
+                for (WatchEvent<?> event : key.pollEvents()) {
+                    WatchEvent.Kind<?> kind = event.kind();
+
+                    if (kind == StandardWatchEventKinds.OVERFLOW) {
+                        continue;
+                    }
+
+                    @SuppressWarnings("unchecked")
+                    WatchEvent<Path> ev = (WatchEvent<Path>) event;
+                    Path filename = ev.context();
+                    Path filePath = LOG_PATH_DIR.resolve(filename);
+
+                    if (kind == StandardWatchEventKinds.ENTRY_CREATE) {
+                        LOG.info("New file detected: " + filePath);
+                        executorService.submit(new LogTask(filePath, stateFilePath));
+                    }
+                }
+
+                boolean valid = key.reset();
+                if (!valid) {
+                    break;
+                }
+            }
+        } catch (IOException e) {
+            LOG.severe("Error watching directory: " + e.getMessage());
+        }
+    }
+
+    private static class LogTask implements Runnable {
+        private final Path filePath;
+        private final String stateFilePath;
+        long filePointer;
+
+        public LogTask(Path filePath, Path stateFilePath) {
+            this.filePath = filePath;
+            this.stateFilePath = stateFilePath.toString() + "/" + filePath.getFileName().toString();
+            this.filePointer = readState();
+        }
+
+        private static String parseLog(String line) {
+            String[] parts = line.split(" ", 7);
+
+            // TODO: better parse
+            return "{" +
+                    "\"traceId\":\"" + parts[0] + "\"," +
+                    "\"spanId\":\"" + parts[1] + "\"," +
+                    "\"operationName\":\"" + parts[5] + "\"," +
+                    "\"references\":[]," +
+                    "\"startTimeUnixNano\":\"" + "1716525641" + "\"," +
+                    "\"endTimeUnixNano\":\"" + "1716525642" + "\"," +
```

Comment:
This part prob needs changing now that the daemon process is sending timestamp-based logs